healthtools_ke_api/build_query.py

```diff
@@ -12,87 +12,76 @@
 
 es = Elastic()
 
+KEYWORDS = {
+        "doctors": ['doc', 'daktari', 'doctor', 'oncologist', 'dr'],
+        "nurses": ['nurse', 'no', 'nursing officer', 'mhuguzi', 'muuguzi', 'RN', 'Registered Nurse'],
+        "clinical-officers": ['co', 'clinical officer', 'clinic officer', 'clinical', 'clinical oficer'],
+        "NHIF": ['nhif', 'bima', 'insurance', 'insurance fund', 'health insurance', 'hospital fund'],
+        "health-facilities": ['hf', 'hospital', 'dispensary', 'clinic', 'hospitali', 'sanatorium', 'health centre']
+        }
 
 class BuildQuery(object):
     def __init__(self):
         self.SMS_RESULT_COUNT = 4  # Number of results to be send via sms
-        self.DOC_KEYWORDS = ['doc', 'daktari', 'doctor', 'oncologist', 'dr']
-        self.CO_KEYWORDS = ['CO', 'clinical officer',
-                            'clinic officer', 'clinical', 'clinical oficer', ]
-        self.NO_KEYWORDS = ['nurse', 'no', 'nursing officer',
-                            'mhuguzi', 'muuguzi', 'RN', 'Registered Nurse']
-        self.NHIF_KEYWORDS = ['nhif', 'bima', 'insurance',
-                              'insurance fund', 'health insurance', 'hospital fund']
-        self.HF_KEYWORDS = ['hf', 'hospital', 'dispensary', 'clinic',
-                            'hospitali', 'sanatorium', 'health centre']
+        
+    def clean_query(self, query): 
+        #removes period from the query, changes the query into a lowercased list '''
+        query = query.replace(".","")
+        query = query.lower().split(" ")
+        return query
 
-    def find_keyword_in_query(self, query, keywords):
-        regex = re.compile(r'\b(?:%s)\b' % '|'.join(keywords), re.IGNORECASE)
-        return re.search(regex, query)
+    def find_keyword(self, query):
+        #finds the keyword in the query
+        names = self.clean_query(query)
+        for key, value in KEYWORDS.items():
+            for name in names:
+                if name in value:
+                    return '{}'.format(key)
+
+    def search_term(self, query):
+        #searches for the name to be searched in the query
+        names = self.clean_query(query)
+        for key, value in KEYWORDS.items():
```

Comment:
DRY: This loop is basically a repetition of the `find_keyword` loop. Instead of running the same loop twice, it'd be more efficient to get the things we need in one run.